From 06f4ab3724b5581c44c94ac240be3dccaaf70338 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 15 Dec 2021 13:47:46 +0200
Subject: [PATCH] [SW-65171] gaudi2: pci: update preset vector at gen3 and 4

In order to avoid TO at phase2, preset vector of Gen3 and Gen4 are
being updated.

Updated preset vector of PCIe Phy to support preset 4,5,6 and 9.

the following sequence should be executed before phy reset
de-assertion:

GEN3_RELATED_OFF = 0x00402001
GEN3_EQ_CONTROL_OFF = 0x05027071
GEN3_RELATED_OFF = 0x01402001
GEN3_EQ_CONTROL_OFF = 0x05027071

Instructed by Haim Revivo.

Change-Id: I0f0a8788564eb02cec6d64bdd53bade200f39781
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 985fe04..2b749f3 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -502,6 +502,14 @@
 	/* Select indirect access to CR PARA I/F instead of PRBI */
 	regmap_writel(1, mmPCIE_PHY_CR_PARA_CFG);
 
+	/* update preset vector at Gen3 to avoid TO at phase2 */
+	regmap_writel(0x00402001, mmPCIE_DBI_GEN3_RELATED_OFF);
+	regmap_writel(0x05027071, mmPCIE_DBI_GEN3_EQ_CONTROL_OFF);
+
+	/* update preset vector at Gen4 to avoid TO at phase2 */
+	regmap_writel(0x01402001, mmPCIE_DBI_GEN3_RELATED_OFF);
+	regmap_writel(0x05027071, mmPCIE_DBI_GEN3_EQ_CONTROL_OFF);
+
 	preboot_init_pcie_phy_fw_update();
 }
 
